Services/WorkflowEngine/classes/activities/class.ilSendMailActivity.php

```diff
@@ -65,13 +65,13 @@ public function execute(): void
         $subject = '';
         foreach ($this->parameters as $parameter) {
             foreach ($definitions as $definition) {
-                if ($definition['id'] = $parameter) {
-                    switch (strtolower($definition['role'])) {
+                if ($definition['id'] == $parameter) {
+                    switch (strtolower((string) ($definition['role'] ?? ''))) {
                         case 'emailaddress':
-                            $recipient = $definition['value'];
+                            $recipient = (string) ($definition['value'] ?? '');
                             break;
                         case 'subject':
-                            $subject = $definition['value'];
+                            $subject = (string) ($definition['value'] ?? '');
```

Comment:
@mbecker-databay Allthough the code does not result in an error anymore after mergin this PR, the `role` attribute is always `NULL`, an will never contain `emailaddress` or `subject`. Please check this afterwards.